pkg/component/operator/web/v0/crawl_website.go

```diff
@@ -128,20 +156,36 @@ func (e *execution) CrawlWebsite(input *structpb.Struct) (*structpb.Struct, erro
 		title := util.ScrapeWebpageTitle(doc)
 		page.Title = title
 
-		defer mu.Unlock()
 		mu.Lock()
 		// If we do not set this condition, the length of output.Pages could be over the limit.
 		if len(output.Pages) < inputStruct.MaxK {
 			output.Pages = append(output.Pages, page)
+
+			// If the length of output.Pages is equal to MaxK, we should stop the scraping.
+			if len(output.Pages) == inputStruct.MaxK {
+				mu.Unlock()
+				cancel()
+				return
+			}
+			mu.Unlock()
+			return
 		}
+		mu.Unlock()
+		cancel()
+
 	})
 
 	// Start scraping
 	if !strings.HasPrefix(inputStruct.URL, "http://") && !strings.HasPrefix(inputStruct.URL, "https://") {
 		inputStruct.URL = "https://" + inputStruct.URL
 	}
-	_ = c.Visit(inputStruct.URL)
-	c.Wait()
+
+	go func() {
+		_ = c.Visit(inputStruct.URL)
```

Comment:
Since it's in a goroutine now. Do we still need to `c.Wait()` here?

Reply:
Yes, we need it. `c.Wait()` is for async crawling in colly package.